nomad/state/state_store.go

```diff
@@ -1885,6 +2164,86 @@ func (s *StateStore) updateSummaryWithJob(index uint64, job *structs.Job,
 	return nil
 }
 
+// updateDeploymentWithAlloc is used to update the deployment state associated
+// with the given allocation
+func (s *StateStore) updateDeploymentWithAlloc(index uint64, alloc, existing *structs.Allocation, txn *memdb.Txn) error {
```

Comment:
Do you want to do this in the state store? Can we provide an update via the Plan instead?